### PR TITLE
1064 - Specific message for GitHub Signup public email

### PR DESCRIPTION
**Pull request by nitishmadhukar**

### CODE REVIEW

File: features/users/create_users.feature [hunk 1]
```diff
@@ -47,9 +47,11 @@ Feature: As a developer
   Scenario: User signs up with a GitHub account having no public email (sad path)
     Given I am on the "registration" page
     When I click "GitHub"
-    Then I should see "Your Github account needs to have a public email address for sign up"
+    Then I should see "Your Github account needs to have a public email address for sign up. Click here for instructions."
     Then I should see the "github-alt" icon
     And I should not see "Password can't be blank"
+    And I click the "Click here for instructions" link
+    Then I should be on ""
```
Comment (tansaku) [inline features/users/create_users.feature]: you don't necessarily have to have the test click through the link to the github page.  VCR could cover that for us by storing the request and then you could check for some relevant contents in the page.

Maybe more importantly we'd just like to be checking for a valid link ...

**Comment by tansaku** [comment]
Great start here @nitishmadhukar - very pleased to see the I18n being used here.  I've made an inline comment about the Cucumber issue you are stuck on.

Some tips to help you improve the GitHub pull request overall:

1) make sure you include a description if possible.  I pulled your code locally, replicated the error, and I think I figured out what you're stuck on, but a verbal overview in the PR description itself is a great back up
2) the PR description should include something like "fixes #1064" so that this pull request is linked to the issue it's trying to solve and so that when we merge this PR the issue will be closed

As regards the code, one thing to consider regarding the Cucumber scenario you are creating is trying to make it more declarative rather than imperative.  You can see this as an advanced option if you are struggling with Cucumber, but ultimately we'd like our scenarios to read as high level descriptions of the system, rather than step by step user experience guides.  There are some blog posts on this topic you might like to read:

* http://itsadeliverything.com/declarative-vs-imperative-gherkin-scenarios-for-cucumber
* http://benmabey.com/2008/05/19/imperative-vs-declarative-scenarios-in-user-stories.html

@mtc2013 and I have just been re-writing the timezone scenarios.  Here's a before and after example:

BEFORE

```gherkin
  Scenario: User in non-UTC timezone Edits an existing event, with no changes, and daylight savings involved
    Given the date is "2015/06/14 09:15:00 UTC"
    And I visit the edit page for the event named "Daily Standup"
    And the user is in "Europe/London" timezone
    Then the start time is "10:00 AM"
    Then the start date is "2014-02-04"
    And I click the "Save" button
    Then I should be on the event "Show" page for "Daily Standup"
    And I should see "09:00-09:30 (UTC)"
    And I visit the edit page for the event named "Daily Standup"
    Then the start time is "09:00 AM"
    And the start date is "2014-02-04"
```

AFTER

```gherkin
  Scenario: User in non-UTC timezone saves an existing event with no changes, during daylight savings
    Given daylight savings are in effect now
    And the user is in "Europe/London"
    And edits an event with start date in standard time
    When they save without making any changes
    Then the event date and time should be unchanged
```

hopefully you can see how the later describes the behaviour we want much more clearly?  If you're unclear about how to get something like this for your current story, don't worry; stick with imperative for now - and we can look at declarative later together.

**Comment by tansaku** [comment]
@nitishmadhukar can you merge this branch to develop?

**Comment by tansaku** [comment]
you might be able to do that through the web interface, or you might need to do locally - I'm seeing a lot of stray commits - you might want to start clean off develop and cherry-pick the relevant commits in ...

**Comment by diraulo** [comment]
@tansaku he has to check merge conflicts on his local machine. Can't do it online..

**Comment by diraulo** [comment]
@tansaku The other commits are probably from earlier on when I updated this PR from the web interface.

**Comment by nitishmadhukar** [comment]
@tansaku I merged the latest upstream branch into this branch in my local. There are no merge conflicts.  I will create a new branch and cherry-pick the required commits and send a new merge request

**Comment by nitishmadhukar** [comment]
close pull request and open another pull request

**Pull request closed without merging**